ironfish/src/wallet/account.ts

```diff
@@ -714,15 +736,25 @@ export class Account {
             'nullifierToNote mappings must have a corresponding decryptedNote',
           )
 
-          await this.walletDb.saveDecryptedNote(
+          const existingTransactionHash = await this.walletDb.getTransactionHashFromNullifier(
             this,
-            noteHash,
-            {
-              ...decryptedNote,
-              spent: false,
-            },
+            spend.nullifier,
             tx,
           )
+          // Remove the nullifier to transaction hash mapping and mark the note as unspent
+          if (existingTransactionHash && existingTransactionHash.equals(transaction.hash())) {
+            await this.walletDb.deleteNullifierToTransactionHash(this, spend.nullifier, tx)
+            await this.walletDb.saveDecryptedNote(
+              this,
+              noteHash,
+              {
+                ...decryptedNote,
+                spent: false,
+              },
+              tx,
+            )
+          }
+
           await this.walletDb.addUnspentNoteHash(this, noteHash, decryptedNote, tx)
```

Comment:
```suggestion

            await this.walletDb.addUnspentNoteHash(this, noteHash, decryptedNote, tx)
          }
```

The `addUnspentNoteHash` call was added in `available-balances`. We should only add the note back into the unspent notes index if we're expiring the only transaction that spent it.

Reply:
Fixed! Good catch